sentry-android-ndk/build.gradle.kts

```diff
@@ -90,12 +89,16 @@ android {
         resolutionStrategy.force(Config.CompileOnly.jetbrainsAnnotations)
     }
 
-    variantFilter {
-        if (Config.Android.shouldSkipDebugVariant(buildType.name)) {
-            ignore = true
-        }
+    buildFeatures {
+        buildConfig = true
+    }
+
+    androidComponents.beforeVariants {
+        it.enable = !Config.Android.shouldSkipDebugVariant(it.buildType)
     }
 
+    ndkVersion = "23.1.7779620"
```

Comment:
nit: I guess we'd also have to bump AGP in sentry-native? Probably will be incompatible after this is merged